### PR TITLE
Upgrade to New SDK

### PR DESCRIPTION
**Pull request by Silvenga**
As many projects in the .Net space, Fleck should target .NetStandard to continue to maintain compatibility with the community. This pull does the following:

- Upgraded primary Fleck project to the new SDK, required by current tooling.
- Changed the target of Fleck to:
  - .Net 4.5
  - .NetStandard 2.0, and
  - .NetCore 2.0
- Changed the supporting projects to target 4.5.

The new SDK tooling does not support 3.5 SP1 at this time. Tracking via https://github.com/Microsoft/msbuild/issues/1333.

### CODE REVIEW

**Comment by Silvenga** [comment]
Two tests were failing before the migration:
- WebSocketServerTests.ShouldStart
- WebSocketServerTests.ShouldSupportDualStackListenWhenServerV4All

**Comment by Silvenga** [comment]
Tests continue to pass under both `net45` and `netcoreapp2.0`, it looks like Fleck is fully compatible with .NetStandard.

**Comment by Silvenga** [comment]
@statianzo I think this pull is complete, aside from the build process updates. Any comments so far?

File: src/Fleck/Fleck.csproj [hunk 1]
```diff
@@ -1,126 +1,12 @@
...
-    <ApplicationVersion>1.0.0.%2a</ApplicationVersion>
-    <IsWebBootstrapper>false</IsWebBootstrapper>
-    <UseApplicationTrust>false</UseApplicationTrust>
-    <BootstrapperEnabled>true</BootstrapperEnabled>
-    <ProductVersion>8.0.30703</ProductVersion>
-    <SchemaVersion>2.0</SchemaVersion>
+    <Title>Fleck</Title>
+    <Authors>statenjason</Authors>
+    <Copyright>Copyright 2018</Copyright>
+    <PackageLicenseUrl>https://github.com/Silvenga/EdiWeave/blob/master/LICENSE</PackageLicenseUrl>
```
Comment (statianzo) [inline src/Fleck/Fleck.csproj]: Can you add the MIT LICENSE file as part of this PR or have this reference http://www.opensource.org/licenses/mit-license.php

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,63 @@
+###############################################################################
```
Comment (statianzo) [inline .gitattributes]: Is adding gitattributes necessary?

File: .gitignore [hunk 3]
```diff
@@ -1,13 +1,218 @@
-bin
-obj
-build
-_ReSharper.*
-*.csproj.user
-*.resharper.user
-*.ReSharper.user
-*.ReSharper
+## Ignore Visual Studio temporary files, build results, and
```
Comment (statianzo) [inline .gitignore]: This is a significant change to the gitignore. Is it necessary?

File: src/Fleck/Fleck.csproj (hunk 1 above)
Comment (Silvenga) [reply src/Fleck/Fleck.csproj]: Oops, copy and paste fail.

File: .gitignore (hunk 3 above)
Comment (Silvenga) [reply .gitignore]: Just using the standard .gitignore for .Net projects provided by VS2017 - same for the .gitattributes. I can remove them if you want - although, I noticed another copy fail on this file...

**Comment by Silvenga** [comment]
Yeah, lots of changes in .NetCore 2.0 that makes a lot of these conversions a lot more easy.

Nuget packaging is generally the same, but using the new SDK tooling over nuget.exe.

The information normally added via the AssemblyInfo can now be passed in via the command line as well.

```
dotnet pack -version 1.0
dotnet nuget push
etc.
```

File: .gitignore (hunk 3 above)
Comment (statianzo) [reply .gitignore]: If it's VS provided, and will make future contributions easier, we can leave it.

**Comment by statianzo** [comment]
It looks fine to me. @AdrianBathurst, do you have any thoughts on this?

**Comment by AdrianBathurst** [comment]
Hi @statianzo , I'm afraid I don't have enough experience with this to give an opinion. Thanks nonetheless.

**Comment by statianzo** [comment]
Alright, thanks for getting back @AdrianBathurst. After I merge, @Silvenga could you assist in publishing to nuget?

**Comment by Silvenga** [comment]
Of course @statianzo, how do you feel about setting up a CI server for building, like Appveyor - and if you want you can securely publish from it.

For example: https://ci.appveyor.com/project/Silvenga/ediweave

**Comment by statianzo** [comment]
That sounds excellent. Fleck previously had CI to run tests and auto-publish to nuget, but the shared teamcity instance it was on has been shut down.

**Comment by statianzo** [comment]
Merged and added you as a collaborator for setting up Appveyor

**Comment by Silvenga** [comment]
Thanks @statianzo, I'll hopefully have some time to set up Appveyor by this weekend.

**Pull request merged**